src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace InterNations\Sniffs\Architecture;
+
+use PHP_CodeSniffer_File as CodeSnifferFile;
+use PHP_CodeSniffer_Sniff as CodeSnifferSniff;
+
+class TestMethodsConventionSniff implements CodeSnifferSniff
+{
+    public function register()
+    {
+        return [T_FUNCTION];
+    }
+
+    public function process(CodeSnifferFile $file, $stackPtr)
+    {
+        $tokens = $file->getTokens();
+
+        $dataProvider = [];
+
+        for ($i = 0; $i < count($tokens); $i++) {
+            if ($tokens[$i]['code'] === T_DOC_COMMENT_OPEN_TAG) {
+                $commentEndPtr = $file->findNext(T_DOC_COMMENT_CLOSE_TAG, ($i + 1));
+
+                for ($j = $i; $j <= $commentEndPtr; $j++) {
+                    if ($tokens[$j]['code'] === T_DOC_COMMENT_TAG && $tokens[$j]['content'] === '@dataProvider') {
+                        $dataProvider[$j] =
+                            trim($tokens[$file->findNext(T_DOC_COMMENT_WHITESPACE, $j + 1, null, true)]['content']);
+                    }
+                }
+            }
+        }
+
+        // Class name
+        $classPtr = $file->findPrevious(T_CLASS, $stackPtr);
+        $className = $tokens[$file->findNext(T_WHITESPACE, $classPtr + 1, null, true)]['content'];
```

Comment:
You can use $file->getIdentifierName($classPtr) here

Reply:
getIdentifierName is not defined.

Reply:
Sorry, I meant getDeclarationName()

Reply:
yes, thank you. later i will try to modify other sniffs as well.